chain/network/src/peer_manager/network_state/tier1.rs

```diff
@@ -92,21 +93,45 @@ impl super::NetworkState {
         let _lock = self.tier1_advertise_proxies_mutex.lock().await;
         let accounts_data = self.accounts_data.load();
 
-        // No matter if this node is a TIER1 node
-
         let vc = match self.tier1_validator_config(&accounts_data) {
             Some(it) => it,
             None => return None,
         };
-        let proxies = match &vc.proxies {
-            config::ValidatorProxies::Dynamic(_) => {
-                // TODO(gprusak): If Dynamic are specified,
-                // it means that this node is its own proxy.
-                // Resolve the public IP of this node using those STUN servers,
-                // then connect to yourself (to verify the public IP).
-                vec![]
+        let proxies = match (&self.config.node_addr, &vc.proxies) {
+            (None, _) => vec![],
+            (_, config::ValidatorProxies::Static(peer_addrs)) => peer_addrs.clone(),
+            // If Dynamic are specified,
+            // it means that this node is its own proxy.
+            // Discover the public IP of this node using those STUN servers.
+            // We do not require all stun servers to be available, but
+            // we require the received responses to be consistent.
+            (Some(node_addr), config::ValidatorProxies::Dynamic(stun_servers)) => {
+                // Query all the STUN servers in parallel.
+                let queries = stun_servers.iter().map(|addr| {
+                    let clock = clock.clone();
+                    let addr = addr.clone();
+                    self.spawn(async move { stun::query(&clock, &addr).await })
+                });
+                let mut node_ips = vec![];
+                for q in queries {
+                    match q.await.unwrap() {
+                        Err(err) => tracing::warn!(target:"network", "STUN lookup failed: {err}"),
```

Comment:
nit: Err should contain the information on which stun server was used.

Reply:
done